rye/tests/test_add.rs

```diff
@@ -163,3 +163,52 @@ fn test_add_flask_wrong_venv_exported() {
     "###);
     fs::remove_dir_all(&fake_venv).unwrap();
 }
+
+#[test]
+fn test_add_from_file_path() {
+    let space = Space::new();
+    space.init("my-project");
+    let fake_venv = space.project_path().join("fake-venv");
+    fs::create_dir_all(&fake_venv).unwrap();
+
+    let wheel_path = get_fixtures_dir()
+        .join("wheels")
+        .join("rye_devtools-1.0.0-py3-none-any.whl");
+    rye_cmd_snapshot!(space.rye_cmd().arg("add").arg("rye_devtools").arg("--path").arg(wheel_path).env("VIRTUAL_ENV", fake_venv.as_os_str()), @r###"
+    
+    "###);
```

Comment:
Need to:
1. Add fixtures filter
2. Maybe use a different wheel for now (see below)

```
program: rye
args:
  - add
  - rye_devtools
  - "--path"
  - /Users/chrispryer/github/rye/rye/tests/fixtures/wheels/rye_devtools-1.0.0-py3-none-any.whl
env:
  RYE_HOME: /Users/chrispryer/github/rye/target/debug/rye-test-dir/home
  __RYE_UV_EXCLUDE_NEWER: "2023-11-18T12:00:00Z"
  VIRTUAL_ENV: /var/folders/x5/94x0_4tj6hz4682vh8l4fv1m0000gn/T/.rye-tests---01Jgvf/project/fake-venv
  UV_CACHE_DIR: /var/folders/x5/94x0_4tj6hz4682vh8l4fv1m0000gn/T/.rye-tests---01Jgvf/uv-cache
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+new results
────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
          0 │+success: false␊
          1 │+exit_code: 1␊
          2 │+----- stdout -----␊
          3 │+Initializing new virtualenv in [TEMP_PATH]/project/.venv␊
          4 │+Python version: cpython@3.12.2␊
          5 │+Added rye_devtools @ file:///Users/chrispryer/github/rye/rye/tests/fixtures/wheels/rye_devtools-1.0.0-py3-none-any.whl as regular dependency␊
          6 │+Reusing already existing virtualenv␊
          7 │+Generating production lockfile: [TEMP_PATH]/project/requirements.lock␊
          8 │+␊
          9 │+----- stderr -----␊
         10 │+  × No solution found when resolving dependencies:␊
         11 │+  ╰─▶ Because only httpx[socks]<0.26.0 is available and rye-devtools==1.0.0␊
         12 │+      depends on httpx[socks]>=0.26.0, we can conclude that␊
         13 │+      rye-devtools==1.0.0 cannot be used.␊
         14 │+      And because only rye-devtools==1.0.0 is available and you require␊
         15 │+      rye-devtools, we can conclude that the requirements are unsatisfiable.␊
         16 │+␊
         17 │+      hint: Pre-releases are available for httpx[socks] in the requested␊
         18 │+      range (e.g., 1.0.0b0), but pre-releases weren't enabled (try:␊
         19 │+      `--prerelease=allow`)␊
         20 │+error: could not write production lockfile for project␊
         21 │+␊
         22 │+Caused by:␊
         23 │+    failed to generate lockfile␊
────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────────

```